plugins/greenhouse/src/dataSources.ts

```diff
@@ -0,0 +1,279 @@
+import type { ManagedCollectionFieldInput } from "framer-plugin"
+import {
+    validateDepartments,
+    validateEducations,
+    validateJobs,
+    validateOffices,
+    validateSections,
+    type Department,
+    type Education,
+    type Job,
+    type Office,
+    type Section,
+} from "./api-types"
+
+export interface GreenhouseDataSource {
+    id: string
+    name: string
+    /**
+     * The fields of the data source.
+     *
+     * The first field is the ID field.
+     * The rest of the fields are the fields of the data source.
+     */
+    fields: readonly GreenhouseField[]
+    apiPath: string
+    fetch: (boardToken: string) => Promise<Job[] | Department[] | Office[] | Education[] | Section[]>
```

Comment:
can we import `GreenhouseItem` and use here like this

```suggestion
    fetch: (boardToken: string) => Promise<GreenhouseItem[]>
```